src/esbmc/bmc.cpp

```diff
@@ -746,7 +749,20 @@ smt_convt::resultt bmct::multi_property_check(
                        &reached_claims,
                        &reached_mul_claims,
                        &total_instance,
-                       &is_goto_cov](const size_t &i) {
+                       &is_goto_cov,
+                       &fail_fast,
+                       &counter](const size_t &i) {
+    //"multi-fail-fast n": stop after first n SATs found.
+    if(!fail_fast.empty() && counter >= stoi(fail_fast))
+    {
+      if(stoi(fail_fast) < 0)
```

Comment:
Two comments:
1. Repeatedly calling `stoi` is not helping performance nor readability.
2. This check whether the `fail_fast` option's value is out of range should not be done repeatedly by the `job_function`, but rather before it. Also: why is this check behind the condition `!fail_fast.empty() && counter >= stoi(fail_fast)`?

Reply:
will move `stoi` and out of range checking outside the function.